xcp_d/utils/confounds.py

```diff
@@ -156,168 +157,16 @@ def load_motion(
-    confoundjs : dict
-        The metadata associated with the confounds file.
-
-    Returns
-    -------
-    pandas.DataFrame
-        The confounds DataFrame, reduced to only include aCompCor regressors.
-    """
-    wm_comp_cor_retained = []
-    csf_comp_cor_retained = []
-    for key, value in confoundjs.items():  # Use the confounds json
-        if "comp_cor" in key and "t" not in key:
-            # Pull out variance explained for white matter masks that are retained
-            if value["Mask"] == "WM" and value["Retained"]:
-                wm_comp_cor_retained.append(key)
-            # Pull out variance explained for CSF masks that are retained
-            if value["Mask"] == "CSF" and value["Retained"]:
-                csf_comp_cor_retained.append(key)
-
-    # grab up to 5 acompcor values
-    N_COLS_TO_GRAB = 5
-
-    # Note that column names were changed from a_comp_cor to w_comp_cor and c_comp_cor
-    # in later versions of fMRIPrep
-    n_wm_comp_cor = N_COLS_TO_GRAB
-    if len(wm_comp_cor_retained) < N_COLS_TO_GRAB:
-        LOGGER.warning(f"Only {len(wm_comp_cor_retained)} white matter CompCor columns found.")
-        n_wm_comp_cor = len(wm_comp_cor_retained)
-
-    # ditto for csf
-    n_csf_comp_cor = N_COLS_TO_GRAB
-    if len(csf_comp_cor_retained) < N_COLS_TO_GRAB:
-        LOGGER.warning(f"Only {len(wm_comp_cor_retained)} CSF CompCor columns found.")
-        n_wm_comp_cor = len(csf_comp_cor_retained)
-
-    acompcor_columns = (
-        wm_comp_cor_retained[:n_wm_comp_cor] + csf_comp_cor_retained[:n_csf_comp_cor]
-    )
-
-    return confounds_df[acompcor_columns]
-
-
-def derivative(confound):
-    """Calculate derivative of a given array.
-
-    Parameters
-    ----------
-    confound : pandas.DataFrame
-        The confound to be modified.
-
-    Returns
-    -------
-    pandas.DataFrame
-        Derivative of the array, with a zero at the beginning.
-        The column(s) will be untitled.
-    """
-    columns = confound.columns.tolist()
-    new_columns = [c + "_dt" for c in columns]
-    data = confound.to_numpy()
-    # Prepend 0 to the differences of the confound data
-    return pd.DataFrame(data=np.diff(data, prepend=0), columns=new_columns)
-
-
-def square_confound(confound):
-    """Square an array.
-
-    Parameters
-    ----------
-    confound : array_like or int or float
-        An array or value to square.
-
-    Returns
-    -------
-    array_like or int or float
-        The squared input data.
-    """
-    columns = confound.columns.tolist()
-    new_columns = [c + "_sq" for c in columns]
-    squared_confounds = confound**2
-    squared_confounds.columns = new_columns
-    return squared_confounds  # Square the confound data
-
-
 @fill_doc
-def load_confound_matrix(params, confound_tsv, custom_confounds=None):
-    """Load a subset of the confounds associated with a given file.
+def load_confound_matrix(params, img_files, custom_confounds=None):
```

Comment:
```suggestion
def load_confound_matrix(params, img_file, custom_confounds=None):
```

Rather than call the parameter "img_files", which implies that it's a list of multiple files, I think you should say "img_file".